selfdrive/car/nissan/interface.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+from cereal import car
+from selfdrive.config import Conversions as CV
+from selfdrive.controls.lib.drive_helpers import create_event, EventTypes as ET
+from selfdrive.controls.lib.vehicle_model import VehicleModel
+from selfdrive.car.nissan.values import CAR
+from selfdrive.car.nissan.carstate import CarState, get_powertrain_can_parser, get_adas_can_parser, get_camera_can_parser
+from selfdrive.car import STD_CARGO_KG, scale_rot_inertia, scale_tire_stiffness, gen_empty_fingerprint
+from selfdrive.car.interfaces import CarInterfaceBase
+
+ButtonType = car.CarState.ButtonEvent.Type
+
+class CarInterface(CarInterfaceBase):
+  def __init__(self, CP, CarController):
```

Comment:
Recently made improvement in car abstraction (see other cars)
the API has changed. The CarState class must be passed in as input.